src/database/arangodb/aql-generator.ts

```diff
@@ -1399,153 +1418,842 @@ register(FollowEdgeQueryNode, (node, context) => {
+              const innerContext = context.introduceVariableAlias(node.itemVariable, itemFrag);
+              return processNode(filterNode, innerContext);
+          }
+        : undefined;
+
+    const outerMapFrag = getFieldTraversalFragment({
+        segments: node.fieldSegments,
+        sourceFrag: rootVar,
+        filterFrag: innerFilterFrag,
+    });
+
+    const itemVar = aql.variable(`item`);
+    const innerContext = context
+        .introduceVariableAlias(node.itemVariable, itemVar)
+        .introduceVariableAlias(node.rootEntityVariable, rootVar);
+
+    // The outerMapFrag will produce a list, so a simple RETURN mapFrag would result in nested lists
+    // -> we iterate over the items again to flatten the lists (the FOR ${itemVar} ...)
+    return aqlExt.parenthesizeList(
+        aql`${forStatementsFrag}`,
+        aql`FOR ${itemVar} IN ${outerMapFrag}`,
+        generateSortAQL(node.orderBy, innerContext),
+        generateLimitClause(node) ?? aql``,
+        aql`RETURN ${node.innerNode ? processNode(node.innerNode, innerContext) : itemVar}`,
+    );
+}
+
+/**
+ * Produces:
+ *
+ *     FOR node1 IN OUTBOUND source_hop1
+ *     FOR node2 in OUTBOUND hop1_hop2
+ *     FOR root IN OUTBOUND hop2_target
+ *     FOR item IN root.fieldSegment1[*].fieldSegment2[**][*
+ *         FILTER filterExpr(CURRENT)
+ *         LIMIT skip, maxCount
+ *         RETURN innerNode(CURRENT)
+ *     ]
+ *
+ * or, if preserveNullValues is true and hop2_target is a to-1 relation:
+ *
+ *     FOR node1 IN OUTBOUND source_hop1
+ *     FOR node2 in OUTBOUND hop1_hop2
+ *     LET root = FIRST(FOR node3 IN OUTBOUND hop2_target RETURN node3)
+ *     FOR item IN root.fieldSegment1[*].fieldSegment2[**][*
+ *         FILTER filterExpr(CURRENT)
+ *         LIMIT skip, maxCount
+ *         RETURN innerNode(CURRENT)
+ *     ]
+ */
+function processTraversalWithRelationAndListFieldSegmentsWithoutSort(
+    node: TraversalQueryNode,
+    context: QueryContext,
+) {
+    if (!node.relationSegments.length || !node.fieldSegments) {
+        throw new Error(`Expected both relation and field segments`);
+    }
+    if (!node.fieldSegments.some((s) => s.isListSegment)) {
+        throw new Error(`Expected at least one list field segment`);
+    }
+    if (!node.orderBy.isUnordered()) {
+        throw new Error(`Did not expect orderBy clauses`);
+    }
+
+    // this will hold the node of the innermost relation traversal
+    const rootVar = aql.variable('root');
+
+    const forStatementsFrag = getRelationTraversalForStatements({
+        node,
+        innermostItemVar: rootVar,
+        preserveNullValues: node.preserveNullValues,
+        context,
+    });
+
+    const innerNode = node.innerNode;
+    const innerMapFrag = innerNode
+        ? (itemFrag: AQLFragment) => {
+              const innerContext = context
+                  .introduceVariableAlias(node.itemVariable, itemFrag)
+                  .introduceVariableAlias(node.rootEntityVariable, rootVar);
```

Comment:
Why is it neccessary to alias the `rootVar` here? It does not look like the inner mapping needs access to the `rootEntityVariable`, but instead only to the current item.